src/Servers/Kestrel/Core/src/Internal/Http2/Http2StreamOfT.cs

```diff
@@ -19,6 +20,7 @@ public Http2Stream(IHttpApplication<TContext> application, Http2StreamContext co
 
         public override void Execute()
         {
+            KestrelEventSource.Log.RequestDequeued(this, AspNetCore.Http.HttpProtocol.Http2);
```

Comment:
I'm thinking of moving this after request parsing. It complicates things a bit though in case that logic fails.

Reply:
We could just invoke Http2Stream.TryValidatePseudoHeaders() before dispatching to the thread pool.

Reply:
I'll do that in another PR.